tokio-threadpool/src/worker/mod.rs

```diff
@@ -396,6 +387,21 @@ impl Worker {
         }
     }
 
+    /// Returns `true` if any worker's queue or the external queue has tasks.
+    fn check_available_tasks(&self) -> bool {
+        if !self.inner.external_queue.is_empty() {
+            return true;
+        }
+
+        for worker in self.inner.workers.iter() {
```

Comment:
TIOLI: I think this might be simpler as:
```rust
    fn check_available_tasks(&self) -> bool {
        if !self.inner.external_queue.is_empty() {
            return true;
        }
        self.inner.workers.iter().any(WorkerEntry::has_tasks)
    }
```